registries/adapters/apiv2_adapter.go

```diff
@@ -262,5 +263,43 @@ func (r APIV2Adapter) loadSpec(imageName string) (*bundle.Spec, error) {
 		registryName = fmt.Sprintf("%s:%s", r.config.URL.Hostname(), r.config.URL.Port())
 	}
 
-	return imageToSpec(body, fmt.Sprintf("%s/%s:%s", registryName, imageName, r.config.Tag))
+	schemaVersion, err := getSchemaVersion(body)
+	if err != nil {
+		return nil, err
+	}
+
+	switch schemaVersion {
+	case 1:
+		log.Debugf("manifest schema 1 for image [%s]", imageName)
+		return responseToSpec(body, fmt.Sprintf("%s/%s:%s", registryName, imageName, r.config.Tag))
+	case 2:
+		log.Debugf("manifest schema 2 for image [%s]", imageName)
+		mConf := manifestConfig{}
+		rdr := bytes.NewReader(body)
+
+		// get the digest
+		err = json.NewDecoder(rdr).Decode(&mConf)
+		if err != nil {
+			log.Errorf("unable to get digest for image [%s]: %v", imageName, err)
+			return nil, err
+		}
+		digest := mConf.Config.Digest
+
+		// get response with digest
+		req, err = r.client.NewRequest(fmt.Sprintf("%s/v2/%s/blobs/%s", r.config.URL, imageName, digest))
+		if err != nil {
+			return nil, err
+		}
+		resp, err = r.client.Do(req)
+		if err != nil {
+			return nil, err
+		}
+		body, err = registryResponseHandler(resp)
+		if err != nil {
+			return nil, fmt.Errorf("%s - error getting configuration object for image [%s] : %s", r.config.AdapterName, imageName, err)
+		}
+		return configToSpec(body, fmt.Sprintf("%s/%s:%s", registryName, imageName, r.config.Tag))
```

Comment:
Is there a reason the parsing for v2 is done inline above, while v1 is all done in a method in adapter.go?

Reply:
For schema v1, we can just read the body, but in schema v2, we need to digest, which is specific for each image, from the 1st response to get the repo config.

Reply:
I think intention was that the other adapters could continue to use `responseToSpec` as they were in the past while schema `2` requires grabbing the `config` section from a different endpoint.